pkg/queue/concurrency_state_test.go

```diff
@@ -278,3 +335,27 @@ func BenchmarkConcurrencyStateProxyHandler(b *testing.B) {
 		reportTicker.Stop()
 	}
 }
+
+// createTempTokenFile creates a temporary file with the text "0123456789" for simulating a serviceAccountToken
+func createTempTokenFile(logger *zap.SugaredLogger, dir string) *os.File {
+	tokenFile, err := os.CreateTemp(dir, "secret")
+	if err != nil {
+		logger.Fatal(err)
+	}
+	if _, err := tokenFile.Write([]byte("0123456789")); err != nil {
+		logger.Fatal(err)
+	}
+	if err := tokenFile.Close(); err != nil {
+		logger.Fatal(err)
+	}
+	return tokenFile
+}
+
+func createConcurrencyEndpoint(e, m, t string) ConcurrencyEndpoint {
```

Comment:
I think I'd convert this in to a NewConcurrencyEndpoint constructor so this is generally useful outside the test (and so consumers of the struct can't get themselves in trouble using it before the token has been stored).